server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/symbols/SymbolAccumulatorService.java

```diff
@@ -52,14 +55,29 @@ public class SymbolAccumulatorService implements VariableAccumulator {
   private final Map<String, SymbolTable> programSymbols;
 
   /** Information related to function definition and references */
-  public static class FunctionInfo {
+  public static class FunctionInfo implements DefinedAndUsedStructure {
```

Comment:
Generally, I don't like this approach. I think that the repository information is ProgramNode specific information. This design allows the information to leak between different compilation units.